pkg/server/api/bundle/v1/service_test.go

```diff
@@ -1079,6 +1079,242 @@ func TestBatchCreateFederatedBundle(t *testing.T) {
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "example.org",
+					},
+				},
+			},
+		},
+		{
+			name:            "Update fails if bundle does not exists",
+			bundlesToUpdate: []*types.Bundle{makeValidBundle(t, federatedTrustDomain)},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{
+					Status: api.CreateStatus(codes.NotFound, "bundle not found"),
+				},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Bundle not found",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         "rpc error: code = NotFound desc = no such bundle",
+					},
+				},
+			},
+		},
+		{
+			name:            "Update datastore query fails",
+			bundlesToUpdate: []*types.Bundle{makeValidBundle(t, federatedTrustDomain)},
+			dsError:         errors.New("datastore error"),
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: api.CreateStatus(codes.Internal, `unable to update bundle: datastore error`)},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Unable to update bundle",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         "datastore error",
+					},
+				},
+			},
+		},
+		{
+			name: "Invalid bundle provided",
+			bundlesToUpdate: []*types.Bundle{
+				{
+					TrustDomain: federatedTrustDomain.String(),
+					X509Authorities: []*types.X509Certificate{
+						{
+							Asn1: []byte("malformed"),
+						},
+					},
+				},
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: api.CreateStatus(codes.Internal, fmt.Sprintf("failed to convert bundle: unable to parse X.509 authority: %v", expectedX509Err))},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Failed to convert bundle",
+					Data: logrus.Fields{
+						telemetry.TrustDomainID: "another-example.org",
+						logrus.ErrorKey:         fmt.Sprintf("unable to parse X.509 authority: %v", expectedX509Err),
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test.logHook.Reset()
+			clearDSBundles(t, test.ds)
+
+			if tt.preExistentBundle != nil {
+				_, err := test.ds.CreateBundle(ctx, &datastore.CreateBundleRequest{
+					Bundle: tt.preExistentBundle,
+				})
+				require.NoError(t, err)
+			}
+
+			test.ds.SetError(tt.dsError)
+
+			resp, err := test.client.BatchUpdateFederatedBundle(context.Background(), &bundlepb.BatchUpdateFederatedBundleRequest{
+				Bundle:     tt.bundlesToUpdate,
+				InputMask:  tt.inputMask,
+				OutputMask: tt.outputMask,
+			})
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+			spiretest.AssertLogs(t, test.logHook.AllEntries(), tt.expectedLogMsgs)
+
+			require.Equal(t, len(tt.expectedResults), len(resp.Results))
```

Comment:
I think you must validate that ds has not updates in error cases, to be sure no call was done to it